src/util/Profiler.h

```diff
@@ -0,0 +1,36 @@
+
+#include <chrono>
+#include <type_traits>
+#include <utility>
+
+namespace util::detail {
+
+/// @brief Profiler function to measure the time consuming
+/// @param func function object, can be a lamdba or function wrapper
+/// @return return a pair if function wrapper has return value: result of
+/// function wrapper and the elapsed time during executing the given function
```

Comment:
What are the units of time?

Reply:
ha. i think it is nanoseconds, will add it in comment~

Reply:
I think would be better to return milliseconds. I think everywhere we log, we convert to milliseconds in one way or another.

Reply:
Good point.